File: solutions/typescript/resistor-color-trio/1/resistor-color-trio.ts
const COLORS = [
  'black', 'brown', 'red', 'orange', 'yellow',
  'green', 'blue', 'violet', 'grey', 'white'
];

export function decodedResistorValue(colors: string[]): string {
  const [c1, c2, c3] = colors;

  const val1 = COLORS.indexOf(c1);
  const val2 = COLORS.indexOf(c2);
  const exponent = COLORS.indexOf(c3);

  let ohms = (val1 * 10 + val2) * Math.pow(10, exponent);

  if (ohms >= 1000000000) {
    return `${ohms / 1000000000} gigaohms`;
  }
  if (ohms >= 1000000) {
    return `${ohms / 1000000} megaohms`;
  }
  if (ohms >= 1000) {
    return `${ohms / 1000} kiloohms`;
  }

  return `${ohms} ohms`;
}

File: solutions/kotlin/resistor-color-trio/1/src/main/kotlin/ResistorColorTrio.kt
import kotlin.math.pow

object ResistorColorTrio {

    fun text(vararg input: Color): String {
     
        val ohms = (input[0].ordinal * 10 + input[1].ordinal) * 10.0.pow(input[2].ordinal).toLong()

        
        return when {
            ohms >= 1_000_000_000 -> "${ohms / 1_000_000_000} gigaohms"
            ohms >= 1_000_000 -> "${ohms / 1_000_000} megaohms"
            ohms >= 1_000 -> "${ohms / 1_000} kiloohms"
            else -> "$ohms ohms"
        }
    }
}
